src/main/java/com/iota/iri/controllers/AddressViewModel.java

```diff
@@ -38,6 +41,34 @@ private AddressViewModel(Address hashes, Indexable hash) {
         this.hash = hash;
     }
 
+    /**
+     * Constructor for an {@link Address} set controller from an existing {@link Address} set.
+     * The resulting hash list is sorted
+     *
+     * @param tangle The tangle reference for the database to find the {@link Address} set in
+     * @param hash The hash identifier for the {@link Address} set that needs to be found
+     * @return The {@link AddressViewModel} controller generated
+     * @throws Exception Thrown if the database cannot load an {@link Address} set from the reference {@link Hash}
+     */
+    public static List<TransactionViewModel> loadAsSortedList(Tangle tangle, Indexable hash) throws Exception {
+        Address hashes = (Address)tangle.load(Address.class, hash);
+        return hashes.set.stream()
+                .map(item -> fromHash(tangle,item))
```

Comment:
When you map with a `fromHash` call as you've done here, a tvm will always be created, which invalidates the `tvm != null` call below when sorting. What happens in this case is that the stream will throw an exception once the sorting hits a transaction that doesn't exist in the db (as is the case when a milestone has been removed from pruning). To fix this issue locally I replaced the mapping with: 
```
.map(
    item -> {
        try {
            if(tangle.exists(Transaction.class, item)) {
            return fromHash(tangle, item);
        } else {
            return null;
        }
    }catch(Exception e){
        return null;
    } 
})
```

This will make sure that if the transaction does not exist, a tvm will not be created for it. Of course this means that you will either need to not include these transactions in the collection before you analyse it in the `LatestMilestoneTrackerImpl` or have line 314 include a null check before trying to add to the queue's. 
i.e 
`if (tvm != null && seenMilestoneCandidates.add(tvm)) {`

Reply:
@achabill  if you apply a change like this the code should pass. As is however it throws an exception the first time it encounters a pruned milestone. Which is the case in the regression test.

Reply:
ok.